### PR TITLE
initial working version of R9M, 8 CH and BINDING support only

### PR DESCRIPTION
**Pull request by UnTraDe**
Added initial support for FrSky R9 protocol using the SX1276 chip.
Currently works with receivers flashed with Flex firmware, and only in 8 CH mode @ 915 MHz.
To enable uncomment the following 2 lines in _Config.h
`//#define SX1276_INSTALLED`
`//#define FRSKYR9_SX1276_INO`

### CODE REVIEW

File: Multiprotocol/FrSkyR9_sx1276.ino [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+
+	const int payload_offset = 8;
+	const bool is_upper = false;
+
+	int chan_index = 0;
+
+	for(int i = 0; i < 8; i += 3)
+	{
+		// map channel values (0-2047) to (64-1984)
+		uint16_t ch1 = 64 + (uint16_t)((1920.0f / 2047.0f) * Channel_data[chan_index]);
```
Comment (pascallanger) [inline Multiprotocol/FrSkyR9_sx1276.ino]: Can you use the existing converstion function instead of floating point calculation: FrSkyX_scaleForPXX()

**Comment by pascallanger** [comment]
Sounds good. Let me know if you can do the small incated modifications. Otherwise I can do it.
Pascal

File: Multiprotocol/FrSkyR9_sx1276.ino (hunk 1 above)
Comment (UnTraDe) [reply Multiprotocol/FrSkyR9_sx1276.ino]: It does the same thing? Should I use it as is or use `convert_channel_frsky` like in the the FrSkyX protocol?

**Comment by UnTraDe** [comment]
@pascallanger I've changed the channel scaling code to use the existing `FrSkyX_scaleForPXX` function. I've also cleaned up the code a bit.

**Comment by pascallanger** [comment]
Thanks, it's in!

**Comment by SILVA79** [comment]
Is it possible to add sx1276 rf module to the atmega328 4in1 multiprotocol module?

**Comment by pascallanger** [comment]
It could be done but you would have to mod your hardware and get rid of one of your RF componenet (CC2500, CYRF, A7105 or NRF)... The STM32 module has been made for this addition in mind and why this is the only supported model.

**Pull request merged**